server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1211,18 +1215,17 @@ static IndexMetadata buildIndexMetadata(
     private void updateIndexMappingsAndBuildSortOrder(
         IndexService indexService,
         CreateIndexClusterStateUpdateRequest request,
-        List<Map<String, Object>> mappings,
+        List<CompressedXContent> mappings,
         @Nullable IndexMetadata sourceMetadata
     ) throws IOException {
         MapperService mapperService = indexService.mapperService();
         IndexMode indexMode = indexService.getIndexSettings() != null ? indexService.getIndexSettings().getMode() : IndexMode.STANDARD;
-        List<Map<String, Object>> mergedMappings = new ArrayList<>(1 + mappings.size());
-        mergedMappings.add(indexMode.getDefaultMapping());
-        mergedMappings.addAll(mappings);
-        for (Map<String, Object> mapping : mergedMappings) {
-            if (mapping.isEmpty() == false) {
-                mapperService.merge(MapperService.SINGLE_MAPPING_NAME, mapping, MergeReason.INDEX_TEMPLATE);
-            }
+        final CompressedXContent defaultMapping = indexMode.getDefaultMapping();
+        if (defaultMapping != null) {
+            mapperService.merge(MapperService.SINGLE_MAPPING_NAME, defaultMapping, MergeReason.INDEX_TEMPLATE);
```

Comment:
I get thrown off here by seeing that you merge in the default mapping, which previously we did not. Why is that?

Reply:
We did merge it previously as well. The only difference is that I stopped needlessly collecting the mappings into a `List`. We'd just merge 2 lines later when iterating the mappings list below.